### PR TITLE
Read scan type from task

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#13361|

## Description

This PR changes the baseline scan making it alerts the vulnerabilities found without considering if they were or were not previously alerted.

## Configuration options

<details>
<summary> Click to expand! </summary>

```xml
  <vulnerability-detector>
    <enabled>yes</enabled>
    <interval>5m</interval>
    <min_full_scan_interval>2m</min_full_scan_interval>
    <run_on_start>yes</run_on_start>

    <!-- Ubuntu OS vulnerabilities -->
    <provider name="canonical">
      <enabled>yes</enabled>
      <os>trusty</os>
      <os>xenial</os>
      <os>bionic</os>
      <os>focal</os>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Debian OS vulnerabilities -->
    <provider name="debian">
      <enabled>yes</enabled>
      <os>stretch</os>
      <os>buster</os>
      <os>bullseye</os>
      <update_interval>1h</update_interval>
    </provider>

    <!-- RedHat OS vulnerabilities -->
    <provider name="redhat">
      <enabled>yes</enabled>
      <os>5</os>
      <os>6</os>
      <os>7</os>
      <os>8</os>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Amazon Linux OS vulnerabilities -->
    <provider name="alas">
      <enabled>no</enabled>
      <os>amazon-linux</os>
      <os>amazon-linux-2</os>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Arch OS vulnerabilities -->
    <provider name="arch">
      <enabled>yes</enabled>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Windows OS vulnerabilities -->
    <provider name="msu">
      <enabled>yes</enabled>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Aggregate vulnerabilities -->
    <provider name="nvd">
      <enabled>yes</enabled>
      <update_from_year>2010</update_from_year>
      <update_interval>1h</update_interval>
    </provider>

  </vulnerability-detector>
```

</summary>
</details>

## Scan-build report

image

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Source upgrade
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report

### CODE REVIEW

**Comment by MiguelazoDS** [comment]
# Manual test
## Environment

## Vulnerabilities found after the first baseline scan

## Vulnerabilities alerted after the first baseline scan

## Vulnerabilities alerted after the second baseline scan (forced by setting the last_full_scan timestamp to 0)

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 1]
```diff
@@ -2593,24 +2588,36 @@ int wm_vuldet_check_agent_vulnerabilities(wm_vuldet_t *vuldet) {
...
+            scan_ctx.scan_type = VU_FULL_SCAN;
+        } else {
+            scan_ctx.scan_type = VU_PARTIAL_SCAN;
+        }
     }
-    else {
-        scan_ctx.scan_type = VU_PARTIAL_SCAN;
-        mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_AG_PART_SCAN, scan_ctx.agent_id);
+
+    switch (scan_ctx.scan_type) {
```
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:2609]: In this case, the `switch` statement isn't necessary.
We could simply write
```
mtdebug1(WM_VULNDETECTOR_LOGTAG, scan_ctx.scan_type, scan_ctx.agent_id);
```

**Review by pereyra-m — COMMENTED** [review]
GJ!

Just one minor comment.
Also, don't forget to update the PR description.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 1 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c]: I implemented the changes as we discussed.

Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c]: It's great now!

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**